lib/xiaomi.js

```diff
@@ -170,7 +170,8 @@ const numericAttributes2Payload = (msg, meta, model, options, dataObject) => {
             payload.mode_switch = {4: 'anti_flicker_mode', 1: 'quick_mode'}[value];
             break;
         case '5':
-            if (['Mains (single phase)', 'DC Source'].includes(meta.device.powerSource) || ['ZNCLDJ12LM'].includes(model.model)) {
+            if (['Mains (single phase)', 'DC Source'].includes(meta.device.powerSource) ||
+                ['ZNCLDJ12LM', 'SJCGQ11LM', 'MFKZQ01LM'].includes(model.model)) {
```

Comment:
Battery-powered devices were excluded from the condition due to very large values of _attribute 5_, which clearly do not correspond to the number of battery removals from the sensor:

- **SJCGQ11LM** (_233_):
```javascript
{"65281":{"1":3055,"10":0,"100":1,"3":35,"4":5032,"5":233,"6":[0,1],"8":516},"modelId":"lumi.sensor_wleak.aq1"}
```

- **MFKZQ01LM** (_4276_):
```javascript
{"65281":{"1":2915,"10":0,"151":0,"152":1,"153":0,"154":4,"3":30,"4":5032,"5":4276,"6":[0,0]},"modelId":"lumi.sensor_cube.aqgl01"}
```

Despite the fact that after removing the battery, the counter increases, it is completely unclear where such large values come from.

Reply:
A possible reason for high _attribute 5_ values for battery-powered sensors is frequent shutdowns due to a nearly dead battery, but this version needs to be tested.

Reply:
I have checked with my devices, I have removed the battery, waited a few seconds, and reconnected it, and the counter has increased by 1 as expected.
Attached the log of the water sensor
I have checked with 2 coordinators, ConBee2/RaspBee2 and zStack3x0 with the latest firmware

<pre>
info  2022-06-16 20:31:45: Coordinator firmware version: '{"meta":{"maintrel":1,"majorrel":2,"minorrel":7,"product":1,"revision":20220219,"transportrev":2},"type":"zStack3x0"}'
info  2022-06-21 00:31:06: Coordinator firmware version: '{"meta":{"maintrel":0,"majorrel":38,"minorrel":120,"product":0,"revision":"0x26780700","transportrev":0},"type":"ConBee2/RaspBee2"}'


debug 2022-06-16 20:32:51: Received Zigbee message from '0x00158d0006c4b533', type 'attributeReport', cluster 'genBasic', data '{"appVersion":6,"modelId":"lumi.sensor_wleak.aq1"}' from endpoint 1 with groupID 0
debug 2022-06-16 20:32:52: Received Zigbee message from '0x00158d0006c4b533', type 'attributeReport', cluster 'genBasic', data '{"65281":{"1":3065,"10":0,"100":0,"3":35,"4":424,"5":8,"6":[0,1],"8":518}}' from endpoint 1 with groupID 0
debug 2022-06-16 20:33:51: Received Zigbee message from '0x00158d0006c4b533', type 'attributeReport', cluster 'genBasic', data '{"65281":{"1":3065,"10":0,"100":0,"3":36,"4":424,"5":8,"6":[0,458752],"8":518},"modelId":"lumi.sensor_wleak.aq1"}' from endpoint 1 with groupID 0
debug 2022-06-16 20:34:18: Received Zigbee message from '0x00158d0006c4b533', type 'attributeReport', cluster 'genBasic', data '{"65281":{"1":3065,"10":0,"100":0,"3":36,"4":5032,"5":8,"6":[0,458752],"8":518},"modelId":"lumi.sensor_wleak.aq1"}' from endpoint 1 with groupID 0
debug 2022-06-16 20:34:43: Received Zigbee message from '0x00158d0006c4b533', type 'attributeReport', cluster 'genBasic', data '{"appVersion":6,"modelId":"lumi.sensor_wleak.aq1"}' from endpoint 1 with groupID 0
debug 2022-06-16 20:34:43: Received Zigbee message from '0x00158d0006c4b533', type 'attributeReport', cluster 'genBasic', data '{"65281":{"1":3065,"10":0,"100":1,"3":36,"4":424,"5":8,"6":[0,1],"8":518}}' from endpoint 1 with groupID 0
debug 2022-06-16 20:35:56: Received Zigbee message from '0x00158d0006c4b533', type 'attributeReport', cluster 'genBasic', data '{"65281":{"1":3065,"10":0,"100":0,"3":36,"4":5032,"5":8,"6":[0,1],"8":518},"modelId":"lumi.sensor_wleak.aq1"}' from endpoint 1 with groupID 0
debug 2022-06-16 20:36:39: Received Zigbee message from '0x00158d0006c4b533', type 'attributeReport', cluster 'genBasic', data '{"65281":{"1":3065,"10":0,"100":0,"3":38,"4":5032,"5":9,"6":[0,0],"8":518},"modelId":"lumi.sensor_wleak.aq1"}' from endpoint 1 with groupID 0
debug 2022-06-16 20:36:48: Received Zigbee message from '0x00158d0006c4b533', type 'attributeReport', cluster 'genBasic', data '{"65281":{"1":3065,"10":0,"100":0,"3":38,"4":5032,"5":9,"6":[0,0],"8":518},"modelId":"lumi.sensor_wleak.aq1"}' from endpoint 1 with groupID 0


debug 2022-06-16 23:10:23: Received Zigbee message from '0x00158d0006c4b533', type 'attributeReport', cluster 'genBasic', data '{"appVersion":6,"modelId":"lumi.sensor_wleak.aq1"}' from endpoint 1 with groupID null
debug 2022-06-16 23:10:23: Received Zigbee message from '0x00158d0006c4b533', type 'attributeReport', cluster 'genBasic', data '{"65281":{"1":3055,"10":0,"100":0,"3":35,"4":424,"5":9,"6":[0,1],"8":518}}' from endpoint 1 with groupID null
debug 2022-06-16 23:11:54: Received Zigbee message from '0x00158d0006c4b533', type 'attributeReport', cluster 'genBasic', data '{"65281":{"1":3055,"10":0,"100":0,"3":35,"4":5032,"5":9,"6":[0,458752],"8":518},"modelId":"lumi.sensor_wleak.aq1"}' from endpoint 1 with groupID null
debug 2022-06-16 23:12:10: Received Zigbee message from '0x00158d0006c4b533', type 'attributeReport', cluster 'genBasic', data '{"appVersion":6,"modelId":"lumi.sensor_wleak.aq1"}' from endpoint 1 with groupID null
debug 2022-06-16 23:12:11: Received Zigbee message from '0x00158d0006c4b533', type 'attributeReport', cluster 'genBasic', data '{"65281":{"1":3055,"10":0,"100":0,"3":35,"4":424,"5":9,"6":[0,1],"8":518}}' from endpoint 1 with groupID null
debug 2022-06-16 23:12:58: Received Zigbee message from '0x00158d0006c4b533', type 'attributeReport', cluster 'genBasic', data '{"65281":{"1":3055,"10":0,"100":0,"3":35,"4":424,"5":9,"6":[0,0],"8":518},"modelId":"lumi.sensor_wleak.aq1"}' from endpoint 1 with groupID null
debug 2022-06-16 23:15:23: Received Zigbee message from '0x00158d0006c4b533', type 'attributeReport', cluster 'genBasic', data '{"65281":{"1":3055,"10":0,"100":0,"3":36,"4":5032,"5":10,"6":[0,0],"8":518},"modelId":"lumi.sensor_wleak.aq1"}' from endpoint 1 with groupID null
debug 2022-06-16 23:16:08: Received Zigbee message from '0x00158d0006c4b533', type 'attributeReport', cluster 'genBasic', data '{"65281":{"1":3055,"10":0,"100":0,"3":37,"4":5032,"5":10,"6":[0,0],"8":518},"modelId":"lumi.sensor_wleak.aq1"}' from endpoint 1 with groupID null
debug 2022-06-16 23:16:13: Received Zigbee message from '0x00158d0006c4b533', type 'attributeReport', cluster 'genBasic', data '{"65281":{"1":3055,"10":0,"100":0,"3":38,"4":5032,"5":10,"6":[0,0],"8":518},"modelId":"lumi.sensor_wleak.aq1"}' from endpoint 1 with groupID null
debug 2022-06-16 23:16:41: Received Zigbee message from '0x00158d0006c4b533', type 'attributeReport', cluster 'genBasic', data '{"65281":{"1":3055,"10":0,"100":0,"3":38,"4":5032,"5":11,"6":[0,0],"8":518},"modelId":"lumi.sensor_wleak.aq1"}' from endpoint 1 with groupID null
debug 2022-06-21 00:31:30: Received Zigbee message from '0x00158d0006c4b533', type 'attributeReport', cluster 'genBasic', data '{"65281":{"1":3065,"10":0,"100":0,"3":33,"4":5032,"5":11,"6":[0,0],"8":518},"modelId":"lumi.sensor_wleak.aq1"}' from endpoint 1 with groupID null
debug 2022-06-21 01:21:36: Received Zigbee message from '0x00158d0006c4b533', type 'attributeReport', cluster 'genBasic', data '{"65281":{"1":3065,"10":0,"100":0,"3":30,"4":5032,"5":11,"6":[0,1],"8":518}}' from endpoint 1 with groupID null
</pre>

Reply:
The fact of the matter is that when the battery is removed, the counter increases and, due to this fact, _attribute 5_ for battery-powered devices is formally `power_outage_count`, but it alerts abnormally high values ​​as in my example.

Accordingly, I see 2 options for _attribute 5_:

- do not add battery-powered devices to exceptions and remove `power_outage_count` expose for them

- completely remove the condition and expose `power_outage_count` for all devices that report this attribute

Let's give @Koenkk a choice.

Reply:
i guess "completely remove the condition and expose power_outage_count for all devices that report this attribute" makes the most sense ( a battery removal is also a power outage)

Reply:
Ok, @inode64 just leave the condition for _attribute 2_, because it is specific, not general.

Reply:
Update